[Issue] FreeCAD segfaults being run with parameters (e.g `.desktop` file has `/usr/bin/freecad --single-instance`)

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Forums discussion

_No response_

### Version

0.20 (Release)

### Full version info

```shell
[code]
OS: Debian GNU/Linux bookworm/sid (GNOME/gnome)
Word size of FreeCAD: 64-bit
Version: 0.20.2.
Build type: Release
Python 3.11.1, Qt 5.15.7, Coin 4.0.0, Vtk 9.1.0, OCC 7.6.3
Locale: Russian/Russia (ru_RU)
[/code]
```


### Subproject(s) affected?

None

### Issue description

FreeCAD segfaults being run with parameters (e.g .desktop file has `/usr/bin/freecad --single-instance`).

```
$ gdb freecad
GNU gdb (Debian 12.1-4+b1) 12.1
Copyright (C) 2022 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from freecad...
Reading symbols from /usr/lib/debug/.build-id/97/3c2c777d1a78a5b6602ce47bea03998897d482.debug...
(gdb) r --single-instance
Starting program: /usr/bin/freecad --single-instance
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
unknown option --single-instance
usage: /usr/bin/freecad [option] ... [-c cmd | -m mod | file | -] [arg] ...
Try `python -h' for more information.
qt.qpa.plugin: Could not find the Qt platform plugin "wayland" in ""
[New Thread 0x7fffee8b86c0 (LWP 1206751)]
[New Thread 0x7fffee0b76c0 (LWP 1206752)]

Thread 1 "freecad" received signal SIGSEGV, Segmentation fault.
__wcslen_sse4_1 () at ../sysdeps/x86_64/multiarch/strlen-sse2.S:142
142	../sysdeps/x86_64/multiarch/strlen-sse2.S: Нет такого файла или каталога.
(gdb) bt
#0  __wcslen_sse4_1 () at ../sysdeps/x86_64/multiarch/strlen-sse2.S:142
#1  0x00007ffff62d0223 in encode_current_locale (text=text@entry=0x0, str=str@entry=0x7fffffffd9f8, error_pos=error_pos@entry=0x0, reason=reason@entry=0x0, 
    raw_malloc=raw_malloc@entry=0, errors=errors@entry=_Py_ERROR_SURROGATEESCAPE) at ../Python/fileutils.c:673
#2  0x00007ffff62d090e in encode_locale_ex (errors=_Py_ERROR_SURROGATEESCAPE, current_locale=0, raw_malloc=0, reason=0x0, error_pos=0x0, str=0x7fffffffd9f8, 
    text=0x0) at ../Python/fileutils.c:818
#3  encode_locale (current_locale=0, raw_malloc=0, error_pos=0x0, text=0x0) at ../Python/fileutils.c:828
#4  Py_EncodeLocale (text=0x0, error_pos=0x0) at ../Python/fileutils.c:855
#5  0x0000555555559034 in main (argc=<optimized out>, argv=0x7fffffffde08) at ./src/Main/MainGui.cpp:252
(gdb) q
A debugging session is active.

	Inferior 1 [process 1206748] will be killed.

Quit anyway? (y or n) y
```

### Anything else?

I **guess** important part is

```
unknown option --single-instance
usage: /usr/bin/freecad [option] ... [-c cmd | -m mod | file | -] [arg] ...
Try `python -h' for more information.
```

`/usr/bin/freecad` is misleading and this is an output from Python invoked [here](https://github.com/FreeCAD/FreeCAD/blob/master/src/Base/Interpreter.cpp#L568-L571). I think, it throws an exception and [`Py_Initialize`](https://github.com/FreeCAD/FreeCAD/blob/1972dfe5a3a2f307804c166c1619e71fefc27283/src/Base/Interpreter.cpp#L580) is not invoked. Then, an exception is caught [here](https://github.com/FreeCAD/FreeCAD/blob/079bdc64b1e4c69a518f4aa80f8b129163c455bc/src/Main/MainGui.cpp#L232), then `Py_GetPath` returns `null` and then `Py_EncodeLocale` segfaults.

-----

Probably this issue is specific to Python < 3.11. A code for older versions [ignores errors and does not throw an exception](https://github.com/FreeCAD/FreeCAD/blob/1972dfe5a3a2f307804c166c1619e71fefc27283/src/Base/Interpreter.cpp#L552).


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

Today I downloaded the files
* http://deb.debian.org/debian/pool/main/f/freecad/freecad_0.20.2+dfsg1-4.dsc
* http://deb.debian.org/debian/pool/main/f/freecad/freecad_0.20.2+dfsg1.orig.tar.xz
* http://deb.debian.org/debian/pool/main/f/freecad/freecad_0.20.2+dfsg1-4.debian.tar.xz
executed `dpkg-source -x freecad_0.20.2+dfsg1-4.dsc` and `debuild`

The compilation went fine but when it came to build the .deb files it stopped with this error:
> Can't exec "/home/user/Downloads/freecad/sources/freecad-0.20.2+dfsg1/debian/freecad-python3.install": No such file or directory at /usr/share/perl5/Debian/Debhelper/Dh_Lib.pm line 1561.
dh_install: error: cannot run debian/freecad-python3.install: No such file or directory
make: *** [debian/rules:3: binary] Error 255

Anyway, when I cd to debian/tmp/usr/share/freecad/bin/ then running `./FreeCAD-Python3 --single-instance` doesn't raise the reported error. So, to me it looks like a packaging problem, not an upstream bug.

--- a/src/Base/Interpreter.cpp
+++ b/src/Base/Interpreter.cpp
@@ -563,7 +563,7 @@ void initInterpreter(int argc,char *argv[])
 {
     PyStatus status;
     PyConfig config;
-    PyConfig_InitPythonConfig(&config);
+    PyConfig_InitIsolatedConfig(&config);
 
     status = PyConfig_SetBytesArgv(&config, argc, argv);
     if (PyStatus_Exception(status)) {

